src/status.ts

```diff
@@ -68,46 +81,96 @@ export class StatusBar implements vscode.Disposable {
         });
     }
 
-    public update() {
-        this.updateSelectedPackage();
-        this.updateBuildButton();
-        this.reloadVisibility();
+    public async update() {
+        await this.updateSelectedPackage();
```

Comment:
Each and every one of these re-resolves the "current package" wouldn't it make more sense to pass it as argument ?

Reply:
Definitely!

Reply:
Updated.